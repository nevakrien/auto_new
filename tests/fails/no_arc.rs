use auto_new::new;

#[derive(new)]
#[no_new_arc]
pub struct Dumby<'a, T>(&'a T);

impl<T: Clone> Dumby<'_, T>{
	pub fn do_stuff(&self) -> T{
		return self.0.clone()
	}
}

fn main(){
	let _ = Dumby::new(&3);
	let _ = Dumby::new_arc(&2);
	println!("hello world");
} 