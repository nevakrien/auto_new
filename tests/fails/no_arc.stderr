error[E0599]: no function or associated item named `new_arc` found for struct `Dumby` in the current scope
  --> tests/fails/no_arc.rs:15:17
   |
5  | pub struct Dumby<'a, T>(&'a T);
   | ----------------------- function or associated item `new_arc` not found for this struct
...
15 |     let _ = Dumby::new_arc(&2);
   |                    ^^^^^^^ function or associated item not found in `Dumby<'_, _>`
   |
note: if you're trying to build a new `Dumby<'_, _>`, consider using `Dumby::<'a, T>::new` which returns `Dumby<'_, _>`
  --> tests/fails/no_arc.rs:3:10
   |
3  | #[derive(new)]
   |          ^^^
   = note: this error originates in the derive macro `new` (in Nightly builds, run with -Z macro-backtrace for more info)
